### ISSUE TITLE
pc_align gets confused when the point clouds spans large planet chunks

### ISSUE THREAD

**Issue by oleg-alexandrov** [NeoGeographyToolkit/StereoPipeline#165]
Dealing with longitude is hard. For some cloud files, it is spans a segment from -180 to 180, for others, some segment between 0 and 360. Sometimes clouds are DEMs, sometimes CSV, and sometimes XYZ. 360 degree adjustments (eg., from -140 degrees to 220 degrees) can be necessary when two different point clouds interact. 

The weird thing is that you can get two point clouds for which the intersection of their lon-lat boxes is not even a single box, but rather two of them. Picture a cloud going from Africa to South America over the Atlantic, and the second doing the same over the Pacific. The intersection of their lon-lat boxes has a chunk in each of the continents, and none of the oceans.

In short, the logic of pc_align, when to comes to finding the shared lon-lat area of two point clouds breaks down for humongous clouds, spanning half a planet or more. The fix will require going very carefully over a lot of code and many testcases. To be done, at some point. 

As an example, consider the longitude segment [-180, 180], and the second one being [0, 360].  They span the whole circle, in different ways. Their intersection is again either one of them, but certainly not [0, 180].

**Comment by oleg-alexandrov** [comment]
There is no quick fix for when a DEM spans most of a planet and is not a usual case either. 

If having the bad luck of encountering such an example it is suggested to either convert the DEMs to some local projection, or use pc_align with --max-displacement -1 when the calculation that finds the shared bounding boxes should be skipped.
